api/apps/graphql/resolvers.js

```diff
@@ -0,0 +1,95 @@
+const User = require('api/apps/users/model.coffee')
+const Curation = require('api/apps/curations/model.coffee')
+const Channel = require('api/apps/channels/model.coffee')
+const Tag = require('api/apps/tags/model.coffee')
+const Author = require('api/apps/authors/model.coffee')
+const { mongoFetch, present, presentCollection, find } = require('api/apps/articles/model/index.coffee')
+
+export const articles = (root, args, req, ast) => {
+  const unpublished = !args.published || args.scheduled
+  if (unpublished && !args.channel_id) {
+    throw new Error('Must pass channel_id to view unpublished articles. Or pass ' +
+      'published: true to only view published articles.'
+    )
+  }
+
+  const unauthorized = !User.hasChannelAccess(req.user, args.channel_id)
+  if (unpublished && unauthorized) {
+    throw new Error('Must be a member of this channel to view unpublished articles. ' +
+      'Pass published: true to only view published articles.'
+    )
+  }
+  return new Promise((resolve, reject) => {
+    mongoFetch(args, (err, res) => {
```

Comment:
Not for this PR, but for the future it'd be good to consider adding to these callback-based APIs so that they return promises. That would allow us to use async/await which could turn this code into...

````
const res = await mongoFetch(args)
resolve(presentCollection(res).results)
````